src/main/java/org/dita/dost/module/reader/AbstractReaderModule.java

```diff
@@ -813,33 +822,44 @@ private String getPrefix(final File relativeRootFile) {
         return res;
     }
 
-    private FileInfo getOrCreateFileInfo(final Map<URI, FileInfo> fileInfos, final URI file) {
+    private void createOrUpdateFileInfo(final URI file, final Consumer<FileInfo> consumer) {
+        for (final FileInfo fi : getOrCreateFileInfo(fileinfos, file)) {
+            consumer.accept(fi);
+        }
+    }
+
+    private Collection<FileInfo> getOrCreateFileInfo(final Map<URI, Collection<FileInfo>> fileInfos, final URI file) {
```

Comment:
Get all `FileInfo` objects by source URI, not just the first (in unstable order).